src/http/content.cr

```diff
@@ -67,11 +67,15 @@ module HTTP
 
       return 0 if @chunk_remaining == 0
 
-      to_read = Math.min(slice.size, @chunk_remaining)
+      to_read = Math.min(count, @chunk_remaining)
 
       bytes_read = @io.read slice[0, to_read]
-      @chunk_remaining -= bytes_read
 
+      if bytes_read == 0
+        raise IO::Error.new("Invalid HTTP chunked content: expected data but got EOF (missing #{@chunk_remaining} more bytes)")
```

Comment:
We should raise an `IO::EOFError` (I forgot we had it). I'm not sure it needs a message: exception class says it's an unexpected EOF, backtrace says where it came from, and I don't think the expected remaining byte count is important.

I.e. best effort to report an invalid protocol/broken communication and allow to deal with it (log, and/or silently close) better than a blunt nil error.